### PR TITLE
[proposal] Extended pattern matching

### PR DESCRIPTION
**Pull request by deemp**
Related to https://github.com/objectionary/normalizer/issues/65#issuecomment-1910968223

Support matching more patterns in PHI expressions.

---

## PR-Codex overview
This PR introduces extended pattern matching syntax to find and extract specific objects deep within a program.

### Detailed summary
- Introduces extended pattern matching syntax for deep object extraction
- Provides Syntax v0 and Syntax v1 examples
- Includes a grammar definition for Syntax v1
- Offers examples and explanations for understanding the syntax and usage

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by deemp** [comment]
@maxonfjvipon, please review

File: proposals/001-extended-pattern-matching.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  matches:
+    - metavar: "?Path"
+      value: ["org", "eolang"]
+    - metavar: "?Name"
+      value: "main"
+    - metavar: "?Object"
+      value: "hello"
+    - metavar: "?Bindings"
+      value: [and ↦ you]
+    - metavar: "?PathHead"
```
Comment (maxonfjvipon) [inline proposals/001-extended-pattern-matching.md:95]: @deemp where did `?PathHead` come from?

Comment (deemp) [reply proposals/001-extended-pattern-matching.md]: From nowhere. Removed in https://github.com/objectionary/normalizer/pull/258/commits/9f3cd7d3c99c9319fcefe89b91a6c6339cc78698

**Comment by deemp** [comment]
This format of requests seems to not be particularly convenient. Closing this for now

**Pull request closed without merging**